File: ManualProg.App/src/app/comments/data-access/responses/comment-response.ts
export interface ICommentResponse {
  id: string;
  content: string;
  repliesCount: number;
  likeCount: number;
  hasLike: boolean;
  profile: ICommentResponseProfileData;
  created: Date;
}

export interface ICommentResponseProfileData {
  id: string;
  name: string;
  hasImage: boolean;
}

File: ManualProg.Api/Features/Comments/Responses/CommentResponse.cs
namespace ManualProg.Api.Features.Comments.Responses;

public record CommentResponse
{
    public required Guid Id { get; set; }
    public required string Content { get; set; }
    public required int RepliesCount { get; set; }
    public required int LikeCount { get; set; }
    public required bool HasLike { get; set; }
    public required ProfileData Profile { get; set; }
    public required DateTime Created { get; set; }

    public record ProfileData
    {
        public required Guid Id { get; set; }
        public required string Name { get; set; }
        public required bool HasImage { get; set; }
    }
}
